components/nexus-httpclient/src/main/java/org/sonatype/nexus/httpclient/config/ConfigurationCustomizer.java

```diff
@@ -88,7 +89,12 @@ private void apply(final ConnectionConfiguration connection, final HttpClientPla
       plan.getClient().setRetryHandler(new StandardHttpRequestRetryHandler(connection.getMaximumRetries(), false));
     }
 
-    // TODO: user-agent (and/or custom headers?)
+    if (connection.getUserAgentCustomisation() != null) {
+      // append UA customisation to one stored as attribute by DefaultsCustomizer
+      plan.getHeaders().put(HttpHeaders.USER_AGENT,
+          plan.getAttributes().get(HttpHeaders.USER_AGENT) + " " + connection.getUserAgentCustomisation()
+      );
+    }
```

Comment:
There are 3 customisers: default, global one, repository one.
if global sets the UA the repository one will not be able to set the UA just by taking the header HttpHeaders.USER_AGENT as it will not be able to overwrite the global value.
eg.
defaults set UA  = Nexus /3.0
global will set UA  = Nexus /3.0 Foo
then if a value is set for repository UA = Bar it will not be able to set the header to Nexus /3.0 Bar